dbschema/budget-record.esdl

```diff
@@ -0,0 +1,13 @@
+module default {
+  type BudgetRecord extending Project::ContextAware {
+    required fiscalYear: int16;
+    required amount: float32 {
+      default := 0;
+    };
+
+    required budget: Budget;
+
+    #TODO
+    # required organization: Organization;
```

Comment:
This should be readonly as well. Let's wait to merge this until org/partners are done. I forgot about this dependency.

Records should be exclusive for budget/project, org, and fiscal year as well.

Reply:
I'm a little unsure what you mean by

> Records should be exclusive for budget/project

Like this?

```
type Record extending Project::Child {
    constraint exclusive on (.project);

Reply:
Something like
```
constraint exclusive on ((.budget, .organization, .fiscalYear))
```
The `budget` is the question there. Idk if that's a stored field or a computed one. Referencing the other unresolved comment below.